components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImplTest.java

```diff
@@ -0,0 +1,434 @@
+                OAuthConstants.AuthorizationCodeState.ACTIVE);
+        String[] tempScope = new String[]{"sms", "email"};
+        persistAuthorizationCodeWithModifiedScope(consumerKey2, authzCodeID2, authzCode2,
+                true, OAuthConstants.AuthorizationCodeState.ACTIVE, tempScope);
+        mockStatic(OAuth2Util.class);
+        when(OAuth2Util.getTenantId(anyString())).thenReturn(DEFAULT_TENANT_ID);
+        when(OAuth2Util.getUserStoreDomain(any())).thenReturn("PRIMARY");
+        mockStatic(OAuth2Util.class);
+        mockStatic(IdentityUtil.class);
+        when(OAuth2Util.getTenantId(anyString())).thenReturn(DEFAULT_TENANT_ID);
+        when(IdentityUtil.isUserStoreInUsernameCaseSensitive(anyString())).thenReturn(true);
+        when(OAuth2Util.isHashDisabled()).thenReturn(true);
+        // Allow the method to pass the validation without wanting to traverse internally.
+        when(OAuth2Util.calculateValidityInMillis(anyLong(), anyLong())).thenReturn(2000L);
+
+        Assert.assertTrue(authorizationCodeDAO.getAuthorizationCodesByUserForOpenidScope(authenticatedUser)
+                .isEmpty());
+        Assert.assertTrue(authorizationCodeDAO.getAuthorizationCodesByUserForOpenidScope(dummyAuthenticatorUser)
+                .isEmpty());
+    }
+
+    @Test(dataProvider = "dataForDoubleAuthzCodePersisting")
+    public void testDeactivateAuthorizationCode(String consumerKey1, String authzCodeID1,
+                                                String authzCode1,
+                                                String consumerKey2, String authzCodeID2,
+                                                String authzCode2) throws Exception {
+
+        AuthzCodeDO authzCodeDO1 = persistAuthorizationCode(consumerKey1, authzCodeID1, authzCode1, true,
+                OAuthConstants.AuthorizationCodeState.ACTIVE);
+        AuthzCodeDO authzCodeDO2 = persistAuthorizationCode(consumerKey2, authzCodeID2, authzCode2, true,
+                OAuthConstants.AuthorizationCodeState.ACTIVE);
+        mockStatic(OAuth2Util.class);
+        mockStatic(IdentityUtil.class);
+        authzCodeDO1.setOauthTokenId(UUID.randomUUID().toString());
+        // If state is EXPIRED/INACTIVE needs to revoke token as well.
+        mockStatic(OAuth2TokenUtil.class);
+        doNothing().when(OAuth2TokenUtil.class, "postRevokeCode", anyString(), anyString(), anyString());
+        authorizationCodeDAO.deactivateAuthorizationCode(authzCodeDO1);
+
+        Assert.assertTrue(authorizationCodeDAO.getActiveAuthorizationCodesByConsumerKey
+                (authzCodeDO1.getConsumerKey()).isEmpty());
+        Assert.assertFalse(authorizationCodeDAO.getAuthorizationCodesByConsumerKey(authzCodeDO2.getConsumerKey()).
+                isEmpty());
+    }
+
+    @Test(dataProvider = "dataForDoubleAuthzCodePersisting")
+    public void testDeactivateAuthorizationCodes(String consumerKey1, String authzCodeID1,
```

Comment:
Would be good if we do not use variable names ending with numbers?

For e.g. consumerKey1 -> consumerKey